### PR TITLE
Refactor 7 Test Files to use makeChoices()

### PR DESCRIPTION
**Pull request by rickyamparo**
Our team refactored 7 test files to use the new makeChoices().  We've maintained the previous flow of the battle, including the player choices for 'idle' pokemon. Please let me know if you have any feedback and we can work to improve it! 

Thank you!

### CODE REVIEW

File: test/simulator/moves/snarl.js [hunk 1]
```diff
@@ -31,9 +32,9 @@ describe('Snarl [Gen 5]', function () {
 			[{species: "Deoxys-Attack", ability: 'victorystar', item: 'laggingtail', moves: ['splash', 'snarl']}],
 			[{species: "Caterpie", level: 2, ability: 'naturalcure', item: 'focussash', moves: ['substitute', 'rest']}],
 		]);
-		battle.commitDecisions();
-		battle.choose('p1', 'move 2');
-		battle.choose('p2', 'move 2');
+
+		battle.makeChoices('move Snarl', '');
```
Comment (Zarel) [inline test/simulator/moves/snarl.js]: This part looks wrong.

It should be:

```
battle.makeChoices('move Splash', 'move Substitute');
battle.makeChoices('move Snarl', 'move Rest');
```

Comment (Zarel) [reply test/simulator/moves/snarl.js]: Remember, when refactoring old code:

`battle.commitDecisions` means "continue onto the next turn, making the default choice for any choices that still need to be made"

`battle.choose` means "make the choice, then, if all choices have been made, continue onto the next turn"

File: test/simulator/moves/belch.js [hunk 2]
```diff
@@ -46,12 +46,10 @@ describe('Belch', function () {
...
 		]);
 		battle.join('p2', 'Guest 2', 1, [{species: 'Registeel', ability: 'prankster', moves: ['rest', 'thunderwave']}]);
-		battle.choose('p2', 'move 2');
-		battle.commitDecisions();
-		battle.choose('p1', 'switch 2'); // For U-Turn
-		battle.choose('p1', 'switch 2');
-		battle.commitDecisions();
-		battle.commitDecisions();
+		battle.makeChoices('move Uturn', 'move Thunderwave');
+		battle.makeChoices('switch 2');//For U-Turn
```
Comment (Zarel) [inline test/simulator/moves/belch.js]: I'd prefer `makeChoices('switch 2', '')`.

Make it clearer which side is making the choice.

**Comment by Zarel** [comment]
If you could remove the `package-lock` changes?

**Comment by Zarel** [comment]
Hello?

**Comment by rickyamparo** [comment]
@Zarel  Sorry about the lack of response, I was taking a break from my computer after graduating from school. 

To clarify some points, you would like to see `makeChoices('switch 2', '')` instead of the u-turn move?

I see what you mean about snarl, I'll go ahead and make those changes. 

I should have these changes made by the end of today and I'll go ahead and push my changes to this PR.

**Comment by Zarel** [comment]
I'm saying to use

    .makeChoices('switch 2', '')

instead of

    .makeChoices('switch 2')

Adding the `, ''` makes it clearer that you're telling player 1 to do the U-turn switch.

**Comment by rickyamparo** [comment]
Awesome! Thank for being so clear in your instructions. I'll get to work on that right away!

**Comment by rickyamparo** [comment]
@Zarel Also, how would I go about removing the package-lock? Do I add it to my .git-ignore?

**Comment by Zarel** [comment]
The mistaken commit was this one:

https://github.com/rickyamparo/Pokemon-Showdown/commit/290ee368cbbe1ae7d8aeb18b0405563df4775c3d

The _easiest_ way to fix this involves removing the `"optional": true` line, and removing the `,` from the previous line, in the `"permessage-deflate"` section, and committing that.

But for the correct way, see my next comment.

**Comment by Zarel** [comment]
The correct way involves actually editing history.

First, interactively edit all your commits so far:

`git rebase --interactive aa2533dc8ec96`

(This is called an interactive rebase - it will walk you through editing all history since the commit ID you pass in.)

What it does is, it gives you a list of commits since then, and asks you what you want to do with them. You can rearrange them, combine them, rename them, etc, all pretty easily.

Unfortunately, what we want to do is the one hard thing: We want to edit a commit. To do that, find the part that says:

> `pick 290ee368 Take out thunderwave`

And change it to

> `edit 290ee368 Take out thunderwave`

(Change "pick" to "edit".)

Now save and close the interactive rebase editor.

You'll now be in edit mode for that commit. Now, restore the previous package-lock:

`git checkout HEAD^ package-lock.json`

(`git checkout` will grab a copy of of a file from somewhere else, `HEAD^` means the previous commit, so this is grabbing the package-lock.json from the previous commit, from before it was changed.)

Now, amend the commit:

`git commit --amend`

Now that you're done with it, you continue the rebase:

`git rebase --continue`

And now you've successfully edited your Git history.

Unfortunately, because we've edited history, we can't just push anymore, we need to force-push. To force-push, use:

`git push --force-with-lease`

**Comment by rickyamparo** [comment]
So i got to the last step of the rebase and I'm getting this error when I try to run git rebase --continue: error: update_ref failed for ref 'refs/heads/test_refactor': cannot lock ref 'refs/heads/test_refactor': is at c81dea1f20d4f93cde748a493d99ed60fd0006b0 but expected e0e513000c3e734819b7080c0afd28fd0e7fbaea
error: could not update refs/heads/test_refactor

Would it be easier if I just closed this PR, delete the branch and start from scratch. I feel like I've fouled this up bad.

**Comment by Zarel** [comment]
No, everything important is still there. I can actually fix it for you if you still need help.

File: test/simulator/abilities/angerpoint.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ describe('Anger Point', function () {
 		const p1 = battle.join('p1', 'Guest 1', 1, [{species: "Cryogonal", ability: 'noguard', moves: ['endure']}]);
 		const p2 = battle.join('p2', 'Guest 2', 1, [{species: "Primeape", ability: 'angerpoint', moves: ['stormthrow']}]);
 
-		battle.commitDecisions();
+		battle.makeChoices('move endude', 'move stormthrow');
```
Comment (Zarel) [inline test/simulator/abilities/angerpoint.js]: >endude

**Pull request merged**